File: go/leetcode/easy/414-第三大的数.go
package main

import (
	"fmt"
	"sort"
)

func thirdMax(nums []int) int {
	sort.Ints(nums)
	fmt.Print(nums)
	b := nums[0]
	m := nums[0]
	s := nums[0]
	for i := 0; i < len(nums); i++ {
		if nums[i] > b {
			s = m
			m = b
			b = nums[i]
			continue
		}
		if nums[i] == b {
			continue
		}

	}
	if m == s {
		return b
	}
	return s
}

func main() {

}

File: python/leetcode/algorithms/easy/414-第三大的数.py
from typing import List


class Solution:
    def thirdMax(self, nums: List[int]) -> int:
        newNums = sorted(set(nums))
        if len(newNums) < 3:
            return newNums[-1]
        else:
            return newNums[-3]


s = Solution()
print(s.thirdMax([1, 1, 2]))
